### ISSUE TITLE
Issues with .TheCallingAssembly() in StructureMap 4.4.2

### ISSUE THREAD

**Issue by jakejgordon** [structuremap/structuremap#526]
For the purposes of this issue, I have 3 projects -- "Web" (which is a .NET MVC 5 project), "IoC" which is a project that contains the StructureMap container registration, and a third project we'll call "Tests". The IoC project has the following class:

```
public static class IoC {
        public static IContainer Initialize() {
            return new Container(c =>
            {
                c.AddRegistry<CommonRegistry>();
                c.AddRegistry<DatabaseRegistry>();
                c.AddRegistry<WebRegistry>();
            });
        }
    }
```

The CommonRegistry is defined as follows:
```
//some other usings
using StructureMap;
using StructureMap.Graph; //<--this is grayed out by the IDE

 public class CommonRegistry : Registry
    {
        public CommonRegistry()
        {
            Scan(
                scan =>
                {
                    scan.TheCallingAssembly();
                    scan.WithDefaultConventions();
                });
       }
}
```

I have included the ```using StructureMap.Graph;``` namespace in CommonRegistry. For some reason Visual Studio has this grayed out in the IDE. Everything compiles fine and when I start my MVC project it loads container and registry and scans everything just fine. No runtime issues.

However, in my Tests project, I create my container in the following manner:
```
 private static readonly IContainer RootContainer = new Container(c =>
        {
            c.AddRegistry<CommonRegistry>();
            c.AddRegistry<DatabaseRegistry>();
        });
```

The call to ```c.AddRegistry<CommonRegistry>() ``` blows up when it tries to ```scan.TheCallingAssembly()```. All three projects reference the same version of StructureMap -- 4.4.2. I've tried cleaning and rebuilding the solutions as well but I can't figure out what's going on.

Any idea what the issue is?

**Comment by jakejgordon** [comment]
I forgot to attach the screenshot of the issue: 
![structuremap_4 4 2_errorjpg](https://cloud.githubusercontent.com/assets/3045786/21087479/1c56cf76-bff4-11e6-9e4c-4063f33e254b.jpg)

**Comment by jeremydmiller** [comment]
Are you developing on Netcoreapp1.0 or Netstandard 1.* something? `TheCallingAssembly` isn't available in SM's CoreCLR support. I'm very surprised that it compiled. Can you just force it by specifying the assembly?

The mechanism that StructureMap has traditionally used to determine the calling assembly isn't supported in Netstandard 1.*.

**Comment by jakejgordon** [comment]
Sorry for the delayed response. This is a .NET 4.5.2 application. Truthfully I didn't notice this (and still don't know how to interpret this), but is [the NuGet page for 4.4.2](https://www.nuget.org/packages/structuremap/4.4.2) says it has dependencies on .NETStandard -- but I'm too much of a newbie to tell from [the nuspec file](https://github.com/structuremap/structuremap/blob/master/src/StructureMap.nuspec) if there are any dependencies that would cause problems for a .NET 4.5.2 application. Furthermore, would Visual Studio/NuGet let me install StructureMap 4.4.2 into a .NET MVC 5 ASP.NET 4.5.2 application if there are dependencies on .NET Standard? I've actually been on StructureMap 4.* since December of 2015 with no problems up until this point.

Sorry for all of the confusion.

**Comment by jeremydmiller** [comment]
@jakejgordon Could I get you to do me a huge favor? Could you drop back to say, SM 4.2, and see if it works there? MissingMethodException pretty well has to mean mismatched assemblies somehow. Any chance there's an errant StructureMap4.dll floating around?

SM 4.4.* targets both .Net 4.5 and Netstandard 1.3/1.5, and all of its dependencies would support either.

**Comment by jakejgordon** [comment]
I certainly can. I'll do it tonight or tomorrow morning and post my
findings back here.

**Comment by jakejgordon** [comment]
First I tried deleting all of my local structuremap* packages from my packages folder and then restoring to see if that would do anything. Notice I've got a few of them:

![all_structuremap_packages](https://cloud.githubusercontent.com/assets/3045786/21168844/341c053a-c185-11e6-95bc-2874cf1dbbdc.jpg)

After that failed, I tried downgrading everything to 4.2.0.402 but I get a NuGet error. I wonder if some files are missing up there now?

![structuremap_4 2 0 402_nuget_error](https://cloud.githubusercontent.com/assets/3045786/21168855/574af41c-c185-11e6-96a5-3953b0bfae1c.jpg)

I then went down an additional level to 4.1.3.394 and it successfully downgraded AND I no longer have the issue with .TheCallingAssembly() being missing!

So... Might be an issue with the latest package? Let me know if there is anything else I should check out.

**Comment by jakejgordon** [comment]
I upgraded back up to version 4.2.* and everything worked fine this time. Maybe the last error was a transient fluke. 

Then I upgraded back up to the latest version 4.4.2 and I am unable to recreate the problem. So there must have indeed been a mismatched assembly _somewhere_ but I'm not sure why it's resolved now. Anyway, I think we can close this issue as I'm unable to recreate it now. If for some reason I got the issue again I could always re-open it.
